os/binfmt/libelf/libelf_addrenv.c

```diff
@@ -153,24 +198,20 @@ int elf_addrenv_alloc(FAR struct elf_loadinfo_s *loadinfo, size_t textsize, size
 #ifdef CONFIG_OPTIMIZE_APP_RELOAD_TIME
 	uint32_t datamemsize = loadinfo->datasize + loadinfo->binp->ramsize;
 	uint32_t rosize = loadinfo->rosize;
+
 	/* loadinfo->datasize contains the size of both data and bss sections.
 	 * But we need to backup only the data section in the RO region. So,
 	 * we need to extend the rosize by just the data section size only. Hence,
 	 * we are subtracting bsssize from loadinfo->datasize.
 	 */
 	loadinfo->rosize += loadinfo->datasize - loadinfo->binp->bsssize;
-#ifdef CONFIG_ARM_MPU
+
 	loadinfo->textsize = 1 << mpu_log2regionceil(0, loadinfo->textsize);
 	loadinfo->rosize = 1 << mpu_log2regionceil(0, loadinfo->rosize);
 	datamemsize = 1 << mpu_log2regionceil(0, datamemsize);
 	loadinfo->binp->ramsize = datamemsize;
```

Comment:
I think we'd better change the variable `ramsize` name.. Because all other's are all for RAM.

Reply:
Initially, i used ramstart and ramsize in case when RELOAD_OPTIMIZE is not defined. In that case, there is a single ram region for entire app. But, in this case, I understand that it does not provide the proper meaning. But, I still used this ramsize variable, because i did not want to introduce new variables in the structure. In future, if we decide to remove the RELOAD_OPTIMIZE config, then i will fix all the variable names properly.